src/popper/runner_host.py

```diff
@@ -293,6 +293,212 @@ def _update_with_engine_config(self, container_args):
+            build = False
+        elif "./" in step.uses:
+            img = f'{pu.sanitized_name(step.id, "step")}'
+            tag = f"{self._config.git_sha_short}"
+            build_ctx_path = os.path.join(self._config.workspace_dir, step.uses)
+        else:
+            _, service, user, repo, step_dir, version = scm.parse(step.uses)
+            wf_cache_dir = os.path.join(self._config.cache_dir, self._config.wid)
+            repo_dir = os.path.join(wf_cache_dir, service, user, repo)
+            img = f"{user}/{repo}".lower()
+            tag = version
+            build_ctx_path = os.path.join(repo_dir, step_dir)
+
+        return (build, img, tag, build_ctx_path)
+
+    def _get_container_kwargs(self, step, img, name):
+        args = {
+            "image": img,
+            "command": list(step.args),
+            "name": name,
+            "volumes": [f"{self._config.workspace_dir}:/workspace",],
+            "working_dir": step.dir if step.dir else "/workspace",
+            "environment": self._prepare_environment(step),
+            "entrypoint": step.runs if step.runs else None,
+            "detach": not self._config.pty,
+            "tty": self._config.pty,
+            "stdin_open": self._config.pty,
+        }
+
+        self._update_with_engine_config(args)
+
+        log.debug(f"container args: {pu.prettystr(args)}\n")
+
+        return args
+
+    def _update_with_engine_config(self, container_args):
+        """Given container arguments, it extends it so it includes options
+        obtained from the popper.config.Config.engine_opts property.
+        """
+        update_with = self._config.engine_opts
+        if not update_with:
+            return
+
+        container_args["volumes"] = [
+            *container_args["volumes"],
+            *update_with.get("volumes", list()),
+        ]
+        for k, v in update_with.get("environment", dict()).items():
+            container_args["environment"].update({k: v})
+
+        for k, v in update_with.items():
+            if k not in container_args.keys():
+                container_args[k] = update_with[k]
+
+    def _create_container(self, cid, step):
+        build, img, tag, build_ctx_path = self._get_build_info(step)
+
+        if build:
+            log.info(f"[{step.id}] podman build {img}:{tag} {build_ctx_path}")
+            if not self._config.dry_run:
+                pass
+        elif not self._config.skip_pull and not step.skip_pull:
+            log.info(f"[{step.id}] podman build {img}:{tag}")
+            if not self._config.dry_run:
+                cmd = ["podman", "pull", f"{img}:{tag}"]
+                HostRunner._exec_cmd(cmd, logging=False)
+
+        if self._config.dry_run:
+            return
+
+        container_args = self._get_container_kwargs(step, f"{img}:{tag}", cid)
+
+        log.debug(f"Container args: {container_args}")
+
+        msg = f"[{step.id}] podman create name={cid}"
+        msg += f' image={container_args["image"]}'
```

Comment:
to be consistent with the code bellow, this can do `msg.append()` instead of `msg +=...`